.github/workflows/onPRReviews.yml

```diff
@@ -1,23 +1,119 @@
 name: On review update
 on:
   pull_request:
-    types: [review_requested, review_request_removed]
+    types: [ready_for_review, review_requested, review_request_removed]
   pull_request_review:
     types: [edited, dismissed, submitted]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  checkMergeReady:
+  get-pr-number:
+    name: Get Current PR Number
+    runs-on: ubuntu-latest
+    outputs: 
+      pr-number: ${{ steps.condval.outputs.value }}
+    steps:
+      - uses: 3drepo/action-cond@v1
+        id: condval
+        with:
+          cond: ${{ github.event_name == 'pull_request' }}
+          if_true: ${{ github.event.number }}
+          if_false: ${{ github.event.pull_request.number }}
+
+  get-issue-number:
+    name: Get Current Issue Number
+    needs: [ get-pr-number ]
+    runs-on: ubuntu-latest
+    outputs: 
+      issue-number: ${{ steps.commitMsgParser.outputs.issue-number }}
+      content-id: ${{ steps.commitMsgParser.outputs.content-id }}
+    steps:
+      - name: Extract Issue Number from PR
+        uses: 3drepo/extract-issue-number-from-PR@v1
+        with:
+          pr: ${{ needs.get-pr-number.outputs.pr-number }}
+          base: staging
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+        id: commitMsgParser
+
+  check-merge-ready:
     name: Ensure all reviewers have approved the PR
     runs-on: ubuntu-latest
+    needs: [ get-issue-number ]
+    outputs: 
+      approved: ${{ steps.checkReviews.outputs.approved }}
     steps:
-      - uses: haya14busa/action-cond@v1
+      - uses: 3drepo/action-cond@v1
         id: condval
         with:
           cond: ${{ github.event_name == 'pull_request' }}
           if_true: ${{ github.event.number }}
           if_false: ${{ github.event.pull_request.number }}
       - name: Check reviews
-        uses: carmenfan/all-reviewers-approved@v1.0
+        id: checkReviews
+        uses: 3drepo/all-reviewers-approved@v1
         with:
           pr: ${{ steps.condval.outputs.value }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check reviews 
+        if: failure()
+        id: setStatusOutputToTrue   
```

Comment:
This is not very easy to read. I'm actually unsure what this is suppose to do doing. Does it stop the script if it's not fully reviewed?

Reply:
always() is because the previous step may fail or succeed based on reviewer approval, I added the addition id: setStatusOutputToTrue to try and stop it erroring when the reviewers aren't ready yet and producing the notification noise all the time.

Reply:
does that mean it fake passes when it fails?